### ISSUE TITLE
Duplicate barcoded items created in the database when duplicates are not allowed.

### ISSUE THREAD

**Issue by klisle** [opensourcepos/opensourcepos#2521]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.2.0

### Issue / Bug  / Question / New Feature

A client using the software described a bug where pressing enter multiple times when adding a new item, created duplicates. I am not sure if this is a real bug but the end result is that there are duplicate  items with the same barcode in the database. As a result editing items is not possible because an error will present, telling you item number is in the database already. Some of the items also do not appear at all when I search for them, regardless of the category I use to filter, even though there may be duplicate instances of the item seen in phpmyadmin. I have considered running two database queries; 1. to remove barcodes from deleted items; 2.  remove barcodes from duplicated not-deleted items and set the deleted field to 1 for those duplicates where the barcode was removed. I wanted to know if just setting the deleted field to 1 is all that is needed or if there was any other way to get rid of the duplicated items. Thanks a lot.

**Comment by jekkos** [comment]
I thought that allowing duplicate barcodes would not make the system unusable. Do you have the configuration option checked? In that case you should not have been allowed to import duplicate barcodes.

On the other hand, if the import did succeed, can you check if the box is ticked? If not please tick it and then see if everything works?

**Comment by jekkos** [comment]
It could indeed be a bug: normally the form submit should be blocked by the browser, but maybe that check is bypassed by hitting enter as that will probably keep the form open and empty the filled in fields.

**Comment by WebShells** [comment]
Jekkos, maybe he meant about creating bulk items, you added the option to insert multiple item by pressing new everytime on item form where it keeps specific fields  there... attributes i guess.

**Comment by jekkos** [comment]
Yes correct that was the functionality I was referring to.

**Comment by daN4cat** [comment]
Merged the fix and I close. If any issue we can reopen this.

**Comment by klisle** [comment]
REOPEN!! issue still persists. getting duplicates when Enter key is rapidly repeatedly pressed when adding new item.

![image](https://user-images.githubusercontent.com/54595880/65986529-a3997700-e449-11e9-8bac-940ae592cf84.png)

**Comment by odiea** [comment]
I do not see where you entered any barcodes.

**Comment by klisle** [comment]
I did not add any barcode. I was just testing adding a new item. Rapidly
pressing the enter button multiple times after filling out the form causes
that item to be duplicated.

**Comment by odiea** [comment]
And in what scenario would that be used?

**Comment by klisle** [comment]
what do you mean? pressing the enter key twice when adding one item
shouldn't give you two entries of the item being created. It should only
register the first Enter press.

**Comment by odiea** [comment]
Rapidly
pressing the enter button multiple times after filling out the form causes
that item to be duplicated.

**Comment by klisle** [comment]
> And in what scenario would that be used?

The scenario in which the original issue was observed was one where 3.2.0 was running on a slow server and the client had people adding items to the system. The users were impatient with the system slowness and at times pressed enter multiple times when adding items, leading to duplicate items, that I as the system engineer am now trying to address before

> Rapidly
> pressing the enter button multiple times after filling out the form causes
> that item to be duplicated.

It's not even that it has to be pressed rapidly but pressing enter multiple times before the new item dialog box goes away will add multiple instances of that same item

**Comment by klisle** [comment]
I was able to duplicate a barcoded item by pressing the enter key twice after filling out the New Item form

**Comment by bkrajendra** [comment]
Release version: 3.3.2 - bb309c (no modification)

I'm still getting this issue while importing Items from CSV.
I guess the issue is related to how an ID is used for barcode generation.

For example, I imported 100+ Items through CSV. I kept barcode fld empty in CSV.
Latter when I generated the barcodes from the Item section (Allow Duplicate Barcodes is unchecked),
I got a duplicate for each ID which is for example 8 and 80, 9 and 90, 7 and 70 and so on...

So now I have a duplicate barcode for every such entry.
for example :
item ID 90 has a barcode 2009000000001
Item ID   9 has a barcode 2009000000001
